daemon/daemon.go

```diff
@@ -652,8 +652,12 @@ func (daemon *Daemon) Shutdown() error {
 	// Keep mounts and networking running on daemon shutdown if
 	// we are to keep containers running and restore them.
 	if daemon.configStore.LiveRestore {
-		return nil
+		// check if there are any running containers, if none we should do some cleanup
+		if ls, err := daemon.Containers(&types.ContainerListOptions{}); len(ls) != 0 || err != nil {
```

Comment:
isn't less expensive to call `daemon.containers.List()` then `daemon.Containers(...)`? I can see much stuff going on with `daemon.Containers(...)` - like filters even if you're providing an empty list option